README.md

```diff
@@ -246,6 +246,66 @@ An example `<selected-recipe-string-key>` would be `"recipe_4.50_bit_mixedpalett
 
 </details>
 
+
+## <a name="using-stable-diffusion-3"></a> Using Stable Diffusion 3
+
+<details>
+  <summary> Details (Click to expand) </summary>
+
+### Model Conversion
+
+Stable Diffusion 3 uses some new and some old models to run. For the text encoders, the conversion can be done using a similar command as before with the `--sd3-version` flag.
+
+```bash
+python -m python_coreml_stable_diffusion.torch2coreml --model-version stabilityai/stable-diffusion-3-medium --bundle-resources-for-swift-cli --convert-text-encoder --sd3-version -o <output-dir>
+```
+
+For the new models (MMDiT, a new VAE with 16 channels, and the T5 text encoder), there are a number of new CLI flags that utilize the [DiffusionKit](https://www.github.com/argmaxinc/DiffusionKit) repo:
+
+- `--sd3-version`: Indicates to the converter to treat this as a Stable Diffusion 3 model
+- `--convert-mmdit`: Convert the MMDiT model
+- `--convert-vae-decoder`: Convert the new VAE model (this will use the 16 channel version if --sd3-version is set)
+- `--include-t5`: Downloads and includes a pre-converted T5 text encoder in the conversion
+
+e.g.:
+```bash
+python -m python_coreml_stable_diffusion.torch2coreml --model-version stabilityai/stable-diffusion-3-medium --bundle-resources-for-swift-cli --convert-vae-decoder --convert-mmdit  --include-t5 --sd3-version -o <output-dir>
+```
+
+To convert the full pipeline with at 1024x1024 resolution, the following command may be used:
+
+```bash
+python -m python_coreml_stable_diffusion.torch2coreml --model-version stabilityai/stable-diffusion-3-medium --bundle-resources-for-swift-cli --convert-text-encoder --convert-vae-decoder --convert-mmdit --include-t5 --sd3-version --latent-h 128 --latent-w 128 -o <output-dir>
+```
+
+Keep in mind that the MMDiT model is quite large and will require increasingly more memory and time to convert as the latent resolution increases.
+
+Also note that currently the MMDiT model requires fp32 and therefore only supports `CPU_AND_GPU` compute units and `ORIGINAL` attention implementation (the default for this pipeline).
+
+### Swift Inference
+
+Swift inference for Stable Diffusion 3 is similar to the previous versions. The only difference is that the `--sd3` flag should be used to indicate that the model is a Stable Diffusion 3 model.
+
+```bash
+swift run StableDiffusionSample <prompt> --resource-path <output-mlpackages-directory/Resources> --output-path <output-dir> --compute-units cpuAndGPU --sd3
+```
+
+### Python MLX Inference
```

Comment:
Thanks for updating this section and the CLI!! 


This section though I feel is still a bit confusing. 
I think it would be good if we follow the same pattern as in [Using Stable Diffusion XL](https://github.com/apple/ml-stable-diffusion?tab=readme-ov-file#-using-stable-diffusion-xl) section. Which is: 
- Model conversion, using the `python_coreml_stable_diffusion.torch2coreml` CLI
- Swift inference, using  `StableDiffusionSample` and the converted Core ML models obtained in step 1
- python inference with `python_coreml_stable_diffusion.pipeline`, with the converted Core ML models obtained in step 1

So for the SD3 section too, if we can follow the same doc structure would be great. In its current state it's confusing that it talks about the converting to Core ML (not MLX) but then in python inference it talks about running it via MLX (not Core ML). 
We already link the to the Diffusion Kit repo in the conversion section, where in the MLX inference can be covered. This repo is mainly focused on generating the Core ML models and running them via python or Swift.

Reply:
Hey @aseemw, we haven't added Python Core ML inference pipelines in this PR. Do you want us to remove the Python MLX inference which is the only Python path at the moment?

Reply:
Yes, that would be great. Would that be a significant extra step?

Reply:
Done! Anything else that is blocking the merge from our side? 🙏

Reply:
> we haven't added Python Core ML inference pipelines in this PR.
Would it possible to add that, in this PR or a subsequent one?

Reply:
Makes sense. Let's get this PR in then!